packages/driver/src/cy/commands/aliasing.ts

```diff
@@ -1,18 +1,34 @@
 import _ from 'lodash'
 import $dom from '../../dom'
+import $errUtils from '../../cypress/error_utils'
 
 export default function (Commands, Cypress, cy) {
-  Commands.addQuery('as', function asFn (alias) {
+  Commands.addQuery('as', function asFn (alias, options = {} as Partial<Cypress.AsOptions>) {
     Cypress.ensure.isChildCommand(this, [alias], cy)
     cy.validateAlias(alias)
 
+    if (!_.isObject(options)) {
+      $errUtils.throwErrByPath('as.invalid_options', { args: { arg: options } })
+    }
+
+    if (options.type && !options.type.match(/query|static/)) {
```

Comment:
```suggestion
    if (options.type && !['query', 'static'].includes(options.type)) {
```

```
'fooqueryfoo'.match(/query|static/) === true
```

Could also change the regex, but this seems like a simpler alternative.